api/envoy/extensions/filters/network/mysql_proxy/v3/mysql_proxy.proto

```diff
@@ -15,15 +21,49 @@ option (udpa.annotations.file_status).package_version_status = ACTIVE;
 // MySQL Proxy :ref:`configuration overview <config_network_filters_mysql_proxy>`.
 // [#extension: envoy.filters.network.mysql_proxy]
 
+// [#next-free-field: 7]
 message MySQLProxy {
   option (udpa.annotations.versioning).previous_message_type =
       "envoy.config.filter.network.mysql_proxy.v1alpha1.MySQLProxy";
 
+  message Route {
+    // String database name that must exact match with client connection db name
+    string database = 1;
+
+    // Upstream cluster to forward the command to.
+    string cluster = 2;
+  }
+
+  message ConnectionPoolSettings {
+    // Max connections exists in the connection pool, default is 10. FIXME(qinggniq)
+    google.protobuf.UInt32Value max_connections = 1;
+
+    // Max spare connections in the connection pool, addition spare connection will be closed, default is 0.
+    google.protobuf.UInt32Value max_idle_connections = 2;
+
+    // Connections when envoy setup, default is 0.
+    google.protobuf.UInt32Value start_connections = 3;
+  }
+
   // The human readable prefix to use when emitting :ref:`statistics
   // <config_network_filters_mysql_proxy_stats>`.
   string stat_prefix = 1 [(validate.rules).string = {min_len: 1}];
 
   // [#not-implemented-hide:] The optional path to use for writing MySQL access logs.
   // If the access log field is empty, access logs will not be written.
   string access_log = 2;
+
+  repeated Route routes = 3;
+
+  ConnectionPoolSettings setting = 4 [(validate.rules).message = {required: true}];
```

Comment:
Can you split out the connection pool aspect of this PR from the split in the auth aspect? This is a very large PR and will be hard to review unless that is done IMHO.